dashboards-observability/public/components/application_analytics/components/application.tsx

```diff
@@ -0,0 +1,244 @@
+      if (isEmpty(queries[tid][RAW_QUERY]) && isEmpty(explorerData[tid])) {
+        emptyTabId = tid;
+        break;
+      }
+    }
+    return emptyTabId;
+  };
+
+
+  const getOverview = () => {
+    return (
+      <Dashboard {...props} page="app" appId="id" appName="Cool Application" />
+    );
+  };
+
+  const getService = () => {
+    return (
+      <Services {...props} page="app" appId="id" appName="Cool Application" />
+    );
+  };
+
+  const getTrace = () => {
+    return (
+      <>
+      <Traces {...props} page="app" appId="id" appName="Cool Application" />
+      <EuiSpacer size='m'/>
+      <SpanDetailPanel
+        {...props}
+        traceId="id"
+        colorMap="color" 
+      />
+    </>
+    );
+  };
+
+  const getLog = () => {
+    return (
+      <LogExplorer 
+        pplService={pplService}
+        dslService={dslService}
+        savedObjects={savedObjects}
+        timestampUtils={timestampUtils}
+        http={http} 
+        history={history} 
+        notifications={notifications} 
+        setToast={setToast}
+        // App Analytics will not be saving queries on Log Events
+        savedObjectId={''} 
+        getExistingEmptyTab={getExistingEmptyTab}
+        />
+    );
+  };
+
+  const getConfig = () => {
+    return (
+      <Configuration />
+    );
+  };
+
+  function getAppAnalyticsTab ({
+    tabId,
+    tabTitle,
+    getContent
+  }: {
+    tabId: string,
+    tabTitle: string,
+    getContent: () => JSX.Element
+  }) {
+    return {
+      id: tabId,
+      name: (<>
+              <EuiText
+                size="s"
+                textAlign="left"
+                color="default"
+              >
+                <span className="tab-title">{ tabTitle }</span>
+              </EuiText>
+            </>),
+      content: (
+        <>
+          { getContent() }
+        </>)
+    };
+  };
+
+  const getAppAnalyticsTabs = () => {
+    return [
+        getAppAnalyticsTab(
+          {
+            tabId: TAB_OVERVIEW_ID,
+            tabTitle: TAB_OVERVIEW_TITLE,
+            getContent: () => getOverview()
+          }
+        ),
+        getAppAnalyticsTab(
+          {
+            tabId: TAB_SERVICE_ID,
+            tabTitle: TAB_SERVICE_TITLE,
+            getContent: () => getService()
+          }
+        ),
+        getAppAnalyticsTab(
+          {
+            tabId: TAB_TRACE_ID,
+            tabTitle: TAB_TRACE_TITLE,
+            getContent: () => getTrace()
+          }
+        ),
+        getAppAnalyticsTab(
+          {
+            tabId: TAB_LOG_ID,
+            tabTitle: TAB_LOG_TITLE,
+            getContent: () => getLog()
+          }
+        ),
+        getAppAnalyticsTab(
+          {
+            tabId: TAB_CONFIG_ID,
+            tabTitle: TAB_CONFIG_TITLE,
+            getContent: () => getConfig()
+          }
+        )
+    ];
+  };
+  
+
+  const memorizedAppAnalyticsTabs = useMemo(() => {
+    return getAppAnalyticsTabs();
+  },
+  []);
```

Comment:
You don't need any dependencies?

Reply:
With no dependencies, it will be called once on mount which is okay

Reply:
But I can also just change it to be used without useMemo, will change in next PR

Reply:
Ideally everywhere we use useMemo, it should have array filled up with dependencies used by its functions/components: https://reactjs.org/docs/hooks-reference.html#usememo

> Conceptually though, that’s what they represent: every value referenced inside the function should also appear in the dependencies array.